sydent/http/httpserver.py

```diff
@@ -99,66 +99,71 @@ def __init__(self, sydent: "Sydent", lookup_pepper: str) -> None:
         identity.putChild(b"api", api)
         identity.putChild(b"v2", v2)
         identity.putChild(b"versions", VersionsServlet())
-        api.putChild(b"v1", v1)
 
-        validate.putChild(b"email", email)
-        validate.putChild(b"msisdn", msisdn)
-
-        validate_v2.putChild(b"email", email_v2)
-        validate_v2.putChild(b"msisdn", msisdn_v2)
-
-        v1.putChild(b"validate", validate)
-
-        v1.putChild(b"lookup", LookupServlet(sydent))
-        v1.putChild(b"bulk_lookup", BulkLookupServlet(sydent))
-
-        v1.putChild(b"pubkey", pubkey)
         pubkey.putChild(b"isvalid", PubkeyIsValidServlet(sydent))
         pubkey.putChild(b"ed25519:0", Ed25519Servlet(sydent))
         pubkey.putChild(b"ephemeral", ephemeralPubkey)
         ephemeralPubkey.putChild(b"isvalid", EphemeralPubkeyIsValidServlet(sydent))
 
+        # v1
+        if not self.sydent.config.general.disable_v1_access:
+            api.putChild(b"v1", v1)
+            validate.putChild(b"email", email)
+            validate.putChild(b"msisdn", msisdn)
+            v1.putChild(b"validate", validate)
+
+            v1.putChild(b"lookup", LookupServlet(sydent))
+            v1.putChild(b"bulk_lookup", BulkLookupServlet(sydent))
+
+            v1.putChild(b"pubkey", pubkey)
+
+            threepid_v1.putChild(b"getValidated3pid", GetValidated3pidServlet(sydent))
+            threepid_v1.putChild(b"unbind", unbind)
+            v1.putChild(b"3pid", threepid_v1)
+
+            email.putChild(b"requestToken", EmailRequestCodeServlet(sydent))
+            email.putChild(b"submitToken", EmailValidateCodeServlet(sydent))
+
+            msisdn.putChild(b"requestToken", MsisdnRequestCodeServlet(sydent))
+            msisdn.putChild(b"submitToken", MsisdnValidateCodeServlet(sydent))
+
+            v1.putChild(b"store-invite", StoreInviteServlet(sydent))
+
+            v1.putChild(b"sign-ed25519", BlindlySignStuffServlet(sydent))
+
+        if (
+            self.sydent.config.general.enable_v1_associations
+            or not self.sydent.config.general.disable_v1_access
+        ):
```

Comment:
I think we probably want to:

1. ensure that `enable_v1_associations` is set to false if `disable_v1_access` is true; and
2. make this only check `enable_v1_associations`